pkg/tree/adl_assembler.go

```diff
@@ -0,0 +1,188 @@
+package tree
```

Comment:
Maybe we can put the ADL stuff into its own folder to make it easier to navigate?

Reply:
yes, I will do this,  and I find that ipld node serializing need n.Length() (at least dagcbor for map node),  so maybe we need add subtreesize( subtree pairs count) field in schema?
<img width="962" alt="image" src="https://user-images.githubusercontent.com/62473385/213413491-708e2f5e-aa37-49e2-946a-3c3370c59c19.png">

Reply:
Oh I see. 🤔 Does it error out if you give it too large a length? @warpfork any thoughts on what we should do about this? Adding a field to track subtree size seems useful since UnixFS does similar, but I wonder if there are any alternatives we could consider.